src/libraries/Microsoft.NETCore.Platforms/readme.md

```diff
@@ -76,80 +76,4 @@ runtimes/win/lib/netstandard1.0/foo.dll
-- `[additional qualifiers]` can be things like `aot`.  Used to further differentiate different platforms.
-
-For all of these we strive to make them something that can be uniquely discoverable at runtime, so that a RID may be computed from an executing application.  As such these properties should be derivable from `/etc/os-release` or similar platform APIs / data.
-
-### Import convention
-Imports should be used when the added RID is considered compatible with an existing RID.
-
-1. Architecture-specific RIDs should first import the architecture-less RID.  EG: `osx.10.11-x64` should first import `osx.10.11`.
-2. Architecture-specific RIDs that are compatible with a previous version RID for the same OS should then import the previous version, architecture specific RID.  EG: `osx.10.11-x64` should then import `osx.10.10-x64`.  If there is no earlier compatible/supported version, then a versionless RID should be imported.  EG: `osx.10.10-x64` should import `osx-x64`.
-3. Architecture-less RIDs that are compatible with a previous version RID for the same OS should then import the previous version, architecture neutral RID.  EG: `osx.10.11` should import `osx.10.10`. If there is no earlier compatible/supported version, then a versionless RID should be imported.  EG: `osx.10.10` should import `osx`.
-4. Version-less RIDs should import an OS category.  EG: `osx-x64` should import `unix-x64`, `osx` should import `unix`.
-
-### Advanced RuntimeGroup metadata
-The following options can be used under special circumstances but break the normal precedence rules we try to establish by generating the RID graph from common logic. These options make it possible to create a RID fallback chain that doesn't  match the rest of the RIDs and therefore is hard for developers/package authors to reason about.  Only use these options for cases where you know what you are doing and have carefully reviewed the resulting RID fallbacks using the CompatibliltyMap.
-
-- `OmitRIDs`: A list of strings delimited by semi-colons that represent RIDs calculated from this RuntimeGroup that should be omitted from the RuntimeGraph.  These RIDs will not be referenced nor defined.
-- `OmitRIDDefinitions`: A list of strings delimited by semi-colons that represent RIDs calculated from this RuntimeGroup that should be omitted from the RuntimeGraph.  These RIDs will not be defined by this RuntimeGroup, but will be referenced: useful in case some other RuntimeGroup (or runtime.json template) defines them.
-- `OmitRIDReferences`: A list of strings delimited by semi-colons that represent RIDs calculated from this RuntimeGroup that should be omitted from the RuntimeGraph.  These RIDs will be defined but not referenced by this RuntimeGroup.
+The RID graphs are frozen and shouldn't be updated anymore. Build from source automatically adds the non-portable distro RID encoded via the `OutputRID` property into the RID graph which allows build tools to target that RID.
```

Comment:
```suggestion
The RID graphs should be only updated with new base OSes. The RID graphs shouldn't be updated with new OS flavor- and version-specific RIDs anymore. Build from source automatically adds the non-portable distro RID encoded via the `OutputRID` property into the RID graph which allows build tools to target that RID.
```

Reply:
Makes sense, thanks. Do we somewhere capture that we don't need to update the RID graph for building an unknown Unix distribution as part of source build? cc @tmds

Reply:
> The RID graphs shouldn't be updated with new OS flavor- and version-specific RIDs anymore.

More specifically: the build automatically takes care of adding the non-portable rid for the OS during source-build.

Reply:
> As the Microsoft.NETCore.Platforms package builds too late, we can't leverage the source built updated RID graph. Therefore we still require manual additions to the RID graph just for the sake of being able to target RIDs in our libraries.

At least for .NET 8, it doesn't seem necessary for the non-portable rid that gets source-built to be in the graph.

> This problem would go away if libraries would use OS runtime detection instead of at build time. Looking at the haiku PR, there aren't just a handful of these libraries. Collapsing build time platforms is general goodness as it reduces the build graph (makes evaluation, restore, build, pack, ... faster).

Yea, we shouldn't create rid specific libs when they are not needed. I imagine most of them exists for rid specific assets, so they are needed.

Haiku is different from adding the non-portable rid like we're doing with source-build.
`haiku` a direct base of `unix`. It's more similar to a portable rid than to a non-portable rid.

Maybe some changes are possible to reduce the verbosity/repetition.

Reply:
> Yea, we shouldn't create rid specific libs when they are not needed. I imagine most of them exists for rid specific assets, so they are needed.

[System.Data.Odbc](https://github.com/dotnet/runtime/blob/cc89f381050ed576f8ca4e527d087b26e7ec6b23/src/libraries/System.Data.Odbc/src/System.Data.Odbc.csproj#L4) is a prime example of a library that targets way too many platforms. I would imagine that all the Unix derived platforms could be collapsed into a single build by using runtime checks (which would then get optimized away when the linker is used in the consuming app).

Reply:
> [System.Data.Odbc](https://github.com/dotnet/runtime/blob/cc89f381050ed576f8ca4e527d087b26e7ec6b23/src/libraries/System.Data.Odbc/src/System.Data.Odbc.csproj#L4) is a prime example of a library that targets way too many platforms. I would imagine that all the Unix derived platforms could be collapsed into a single build by using runtime checks (which would then get optimized away when the linker is used in the consuming app).

I took a quick look. It seems the split is mostly for finding the platform specific name of the Odbc32 library. Perhaps we can do something with `NativeLibrary` instead. @ViktorHofer may be you can create an issue to investigate it further.

Reply:
Exactly. ODBC is tracked via https://github.com/dotnet/runtime/issues/53900 but there are probably also inbox libraries that would benefit from a build platform simplification.